clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/loadbalancer/UpsertAmazonLoadBalancerAtomicOperation.groovy

```diff
@@ -154,4 +174,62 @@ class UpsertAmazonLoadBalancerAtomicOperation implements AtomicOperation<UpsertA
     task.updateStatus BASE_PHASE, "Done deploying load balancers."
     operationResult
   }
+
+  private String updateApplicationSecurityGroupIngressRules(List<Listener> listeners) {
+    Names names = Names.parseName(description.name)
+    String application = names.getApp()
+    final securityGroupLookup = securityGroupLookupFactory.getInstance(description.region)
+    def securityGroupUpdater = securityGroupLookup.getSecurityGroupByName(
+      description.credentialAccount,
+      application,
+      description.vpcId
+    ).orElse(null)
+
+    if (!securityGroupUpdater) {
+      log.info("Application security group not found for $application")
+      return null
+    }
+
+    try {
+      List<IpPermission> currentPermissions = SecurityGroupIngressConverter.flattenPermissions(securityGroupUpdater.securityGroup)
+      List<IpPermission> permissionsFromListeners = listeners.collect {
+        new IpPermission(
+          ipProtocol: "tcp",
+          fromPort: it.getInstancePort(),
+          toPort: it.getInstancePort(),
+          ipv4Ranges: Collections.singletonList(new IpRange(cidrIp: "0.0.0.0/0"))
```

Comment:
Is the goal of this PR to ingress the application elb security group into the application security group?

In that case I suspect you should set `userIdGroupPairs` on the `IpPermission`, believe that's what we're normally doing.